reedstrm/baking_workflow_and_errors.md

```diff
@@ -0,0 +1,120 @@
+# Baking and Recipes
+
+Baking has become a critical step for publishing - however, it is not planned for it to cause publishing to fail. This is partly due to how long baking may take - much like PDF generation, it can take several hours, if there are external parts, like exercises, that need to be fetched and incorporated into the pages.
+
+This slightly odd situation (is baking a publish blocker, or not?) comes about partly due to derived copies - publishing a new version of content impacts not only the immediate book, but any book that shares pages with it. It was decided that failure of related books to bake is _not_ a publication blocker, so by extension, it is not for any book. However, we want to be careful to not break existing books by publishing new content, but we cannot test every book that contains a given page.
```

Comment:
No, it's not about the hanging request. It's going to be async at some point in time. In fact, this is how Press intends to do publications as well.

The issue is whether a book can be in a pre-published state. Technically, that would be entries in the `modules` table, but not yet in the `latest_modules` table. If a _tobe published_ page is shared within multiple books, those books that use the page can fail publication due to baking issues. Thus those publications result in a pre-publication state. That is they use a valid version number, but are not fully-published (entered in the `latest_modules` table).

It's then, either the recipe author's job to adjust the recipe to work for all books or the content author's job to revise the book to bring it up-to-speed; and therefore resolving baking issues. I'm looking at this similar to how one would resolve merge conflicts in VCS. Or another apt scenario would be how conflicts are resolved in dependency package management. Either case, puts the responsibility on the original author to resolve the conflict.

I must have misunderstood this, but I believe I've come around to understanding it as long as all the pre-publication state and fully-published state business makes sense.